main/src/library/File.flix

```diff
@@ -309,19 +309,29 @@ namespace File {
             import new java.lang.String(Array[Int8, Static], ##java.nio.charset.Charset): String & Impure as newString;
 
             let javaFile = newFile(f);
-            let stream = newInputStream(javaFile);
-            discard skip(stream, opts.offset);
-            let bytes = [0i8; opts.count];
-            discard read(stream, bytes, 0, opts.count);
-            let charSet = forName(opts.charSet);
-            let returnString = newString(bytes, charSet);
-            Ok(returnString)
+            def loop(i) = {
+                if (0 <= i) {
+                    let stream = newInputStream(javaFile);
+                    let skipped = skip(stream, opts.offset);
+                    if (skipped == opts.offset) {
+                        let bytes = [0i8; opts.count];
+                        discard read(stream, bytes, 0, opts.count);
+                        let charSet = forName(opts.charSet);
+                        let returnString = newString(bytes, charSet);
+                        Ok(returnString)
+                    } else {
```

Comment:
In this case some `bytes < opts.offset` might have been skipped, which means that the next loop should only try to skip `opts.offset-bytes`. `loop` should probably have a `bytesToSkip` argument that can be decreased.